tutorials/build_system/gradle_guide.md

```diff
@@ -0,0 +1,157 @@
+
+If a matching version of the specified IntelliJ Platform is not available on the local machine, the Gradle plugin downloads the correct version and type.
+IntelliJ IDEA then indexes the build and any associated source code and JetBrains Java Runtime.
+
+#### IntelliJ Platform Configuration
+Explicitly setting the [Setup DSL](https://github.com/jetbrains/gradle-intellij-plugin#setup-dsl) attributes `intellij.version` and `intellij.type` tells the Gradle plugin to use that configuration of the IntelliJ Platform to build the plugin project.
+If a local installation of IntelliJ IDEA is the desired type and version of the IntelliJ Platform, use `intellij.localPath` to point to that installation.
+If the `intellij.localPath` attribute is set, do not set the `intellij.version` and `intellij.type` attributes as this could result in undefined behavior.
+
+#### Plugin Dependencies
+IntelliJ Platform plugin projects may depend on either bundled or third-party plugins.
+In that case, a project should build against a version of those plugins that match the IntelliJ Platform version used to build the plugin project.
+The Gradle plugin will fetch any plugins in the list defined by `intellij.plugins`.
+See the Gradle plugin [README](https://github.com/jetbrains/gradle-intellij-plugin#setup-dsl) for information about specifying the plugin and version.
+
+Note that this attribute describes a dependency so the Gradle plugin can fetch the required artifacts.
+The IntelliJ Platform plugin project is still required to declare these dependencies in its [Plugin Configuration](/basics/plugin_structure/plugin_configuration_file.md) (`plugin.xml`) file.
+
+### Configuring the Gradle Plugin for Running IntelliJ Platform Plugin Projects
+By default, the Gradle plugin will use the same version of the IntelliJ Platform for the IDE Development Instance as was used for building the plugin.
+Using the corresponding JetBrains Runtime is also the default, so for this use case no further configuration is required.
+
+#### Running Against Alternate Versions and Types of IntelliJ Platform-Based IDEs
+The IntelliJ Platform IDE used for the Development Instance can be different from that used to build the plugin project.
+Two Gradle plugin attributes can be set to the path of an alternate (different version or non-IDEA) IntelliJ Platform-based IDE to control the configuration used for the IDE Development Instance:
+* Setting the [Running DSL](https://github.com/jetbrains/gradle-intellij-plugin#running-dsl) attribute `runIde.ideaDirectory` will define an IDE to be used for the Development Instance in that single `runIde` task.
+* Setting the Setup DSL attribute `intellij.alternativeIdePath` will define an IDE as the default value for all `runIdea.ideaDirectory` attributes of all `runIde` tasks, including `runIde.buildSearchableOptions`.
```

Comment:
I'd omit mentioning of `alternativeIdePath` property. Cannot think of any scenario where it could be useful

Reply:
I'm confused about using `intellij.alternativeIdePath`. The first note for the attribute in the [README](https://github.com/JetBrains/gradle-intellij-plugin#rundebug-ide-properties) says _Use this property if you want to test your plugin in any non-IDEA JetBrains IDE such as WebStorm or Android Studio._ 

So I thought that was the primary use case, although in greater scope than `runIde.ideaDirectory`. 

Consequently, I used `intellij.alternativeIdePath` as the basis for the code snippet in the section _Plugins Targeting Alternate IntelliJ Platform-Based IDEs_, which targets PhpStorm.  

Is the rule that `intellij.alternativeIdePath` should only be used for plugin projects that have multiple `runIde` tasks?

Reply:
> Is the rule that intellij.alternativeIdePath should only be used for plugin projects that have multiple runIde tasks?

There is no such rule, as some other `intellij` extension properties, that one was added way before `runIde` task became configurable, so it just kind of legacy.